packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -714,7 +710,7 @@ export const useGridColumnResize = (
    * EFFECTS
    */
 
-  React.useEffect(() => stopListening, [stopListening]);
+  useEnhancedEffect(() => stopListening, [stopListening]);
```

Comment:
In case the component unmounts while the user is resizing, remove the listener synchronously.